datacube/model/__init__.py

```diff
@@ -460,8 +463,27 @@ def fix_nodata(m):
                     m['nodata'] = float(nodata)
                 return m
 
-            self._canonical_measurements = OrderedDict((m['name'], Measurement(**fix_nodata(m)))
-                                                       for m in self.definition.get('measurements', []))
+            self._canonical_measurements = OrderedDict()
+            for m in self.definition.get('measurements', []):
+                # Generate 2D measurements from a 3D measurement so we dont have to fully enumerate them
+                # in the product definition
+                # Note: Might be a good idea to natively support 3D measurements
+                # Todo: See if we can make a sister change to eo3 to describe a 3D datasource
+                '''
+                if 'extra_dim' in m:
+                    for idx, mm in enumerate(m['extra_dim'].get('measurement_map', [])):
+                        new_m = deepcopy(m)
+                        new_m.update({'name': mm})
+                        new_m.update({'extra_dim_index': idx})
+                        if 'alias_map' in m['extra_dim']:
+                            new_m.update({'alias': m['extra_dim']['alias_map'][idx]})
+                        if 'spectral_definition_map' in m['extra_dim']:
+                            new_m.update({'spectral_definition': m['extra_dim']['spectral_definition_map'][idx]})
+                        new_m.pop('extra_dim', None)
+                        self._canonical_measurements.update({new_m['name']: Measurement(**fix_nodata(new_m))})
+                '''
```

Comment:
Let's not keep commented out code.

Reply:
Will do.

Reply:
Actually this can go back to the original code in develop.

Reply:
Done, Please let me know if there are any other required changes.